k8s/service-account/main.tf

```diff
@@ -0,0 +1,30 @@
+resource "kubernetes_namespace_v1" "main" {
+  metadata {
+    name   = var.namespace
+    labels = var.labels
+  }
+}
+
+resource "kubernetes_service_account_v1" "main" {
+  metadata {
+    name      = "massdriver-cloud-provisioner"
+    labels    = var.labels
+    namespace = kubernetes_namespace_v1.main.metadata.0.name
+  }
+}
+
+resource "kubernetes_secret_v1" "main" {
+  metadata {
+    name      = "massdriver-cloud-provisioner"
+    namespace = kubernetes_namespace_v1.main.metadata.0.name
+    annotations = {
+      "kubernetes.io/service-account.name" = kubernetes_service_account_v1.main.metadata.0.name
+    }
+  }
+  type = "kubernetes.io/service-account-token"
+
+  # Without this there's a race condition that's harddd to debug
+  # I think how it works is that the resource returns and downstream
+  # things trying to use the token fail validation. Token can't be xyz...
+  wait_for_service_account_token = true
+}
```

Comment:
Missing the `kubernetes_cluster_role_binding_v1`

Reply:
That happens in the gke-core-services module. I did the binding there. This module produces a k8s service account, and downstream modules can bind

Reply:
You can even see, in the first version of this [PR](https://github.com/massdriver-cloud/terraform-modules/pull/97/commits/68ea3345f5126a1460a6ed31e6bfaa38f5314f5a#diff-b5817e9f59ccbc250c2092ba6cbe7ee108e7b879258a7d6f394dd4a29ec536d1R7) where I do the binding in the downstream resource. 

I backed that out because of the changes you merged, there were conflicts at that point. But I believe this is the right way to model this module. It makes a SA, binding happens elsewhere. LMK what you think. :)